pkg/cmd/checkendpoint/cmd.go

```diff
@@ -0,0 +1,38 @@
+package checkendpoint
+
+import (
+	"context"
+	"os"
+	"time"
+
+	operatorcontrolplaneclient "github.com/openshift/client-go/operatorcontrolplane/clientset/versioned"
+	operatorcontrolplaneinformers "github.com/openshift/client-go/operatorcontrolplane/informers/externalversions"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/cmd/checkendpoint/controller"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/version"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/spf13/cobra"
+)
+
+func New() *cobra.Command {
+	config := controllercmd.NewControllerCommandConfig("check-endpoint", version.Get(), func(ctx context.Context, cctx *controllercmd.ControllerContext) error {
+		operatorcontrolplaneClient := operatorcontrolplaneclient.NewForConfigOrDie(cctx.KubeConfig)
+		operatorcontrolplaneInformers := operatorcontrolplaneinformers.NewSharedInformerFactoryWithOptions(operatorcontrolplaneClient, 10*time.Minute, operatorcontrolplaneinformers.WithNamespace("openshift-kube-apiserver"))
+		check := controller.New(
+			os.Getenv("POD_NAME"),
+			"openshift-kube-apiserver",
+			operatorcontrolplaneClient.ControlplaneV1alpha1(),
+			operatorcontrolplaneInformers.Controlplane().V1alpha1().PodNetworkConnectivityChecks(),
+			cctx.EventRecorder,
+		)
+		operatorcontrolplaneInformers.Start(ctx.Done())
+		check.Run(ctx, 1)
+		<-ctx.Done()
+		return nil
+	})
+	config.DisableServing = true
+	config.Config()
+	cmd := config.NewCommandWithContext(context.Background())
+	cmd.Use = "check-endpoint"
+	cmd.Short = "Checks that a tcp connection can be opened to an endpoint."
```

Comment:
nit: falling over the singular use of "endpoint" everywhere in the PR. Isn't this checking a number of them potentially?

Reply:
Done. Reviewed the plurality of endpoint where used and made adjustments as needed to clarify.